src/components/skills-quiz/TopSkillsOverview.jsx

```diff
@@ -1,32 +1,38 @@
-import {
-  useContext, useEffect, useState, useMemo,
-} from 'react';
+import { useContext, useMemo } from 'react';
 import PropTypes from 'prop-types';
-import { Badge, Card } from '@openedx/paragon';
+import { Badge, Card, Skeleton } from '@openedx/paragon';
 import { SearchContext } from '@edx/frontend-enterprise-catalog-search';
+import { getConfig } from '@edx/frontend-platform/config';
+import { Configure, InstantSearch } from 'react-instantsearch-dom';
 import { SkillsContext } from './SkillsContextProvider';
 import { DROPDOWN_OPTION_IMPROVE_CURRENT_ROLE } from './constants';
 import SelectedJobSkills from './SelectedJobSkills';
-import SimilarJobs from './SimilarJobs';
 import JobDescriptions from './JobDescriptions';
+import { isObjEmpty, useAlgoliaSearch } from '../app/data';
+import { AlgoliaFilterBuilder } from '../AlgoliaFilterBuilder';
+import CardLoadingSkeleton from './CardLoadingSkeleton';
+import SimilarJobs from './SimilarJobs';
+
+import { withCamelCasedStateResults } from '../../utils/HOC';
 
 const getJobSkills = (job) => {
   const jobSkills = job[0]?.skills?.sort((a, b) => (
     (a.significance < b.significance) ? 1 : -1));
   return jobSkills?.slice(0, 7);
 };
 
-const TopSkillsOverview = ({ index }) => {
+const TopSkillsHits = ({ hits, isLoading }) => {
   const {
     state: {
       selectedJob, goal, currentJobRole, interestedJobs,
     },
   } = useContext(SkillsContext);
+
   const { refinements: { current_job: currentJob, industry_names: industryNames } } = useContext(SearchContext);
-  const [currentJobDetails, setCurrentJobDetails] = useState(null);
   const jobSelected = goal === DROPDOWN_OPTION_IMPROVE_CURRENT_ROLE ? currentJobRole : interestedJobs;
+
   const selectedJobDetails = useMemo(
-    () => jobSelected?.filter(job => job.name === selectedJob) || [],
+    () => jobSelected?.filter(job => job?.name === selectedJob) || [],
```

Comment:
[nit/curious]: is the added `?.` in `job?.name` needed? If `jobSelected` is defined, is it possible for this array to contain `undefined`/`null` (e.g., `[undefined, null]`)? If not, the `?.` could likely be removed. Otherwise, that might be a symptom of a larger problem (e.g., how to prevent `jobSelected` array from containing `undefined`/`null`)